auth/api/http/groups/requests.go

```diff
@@ -0,0 +1,146 @@
+package groups
+
+import (
+	"github.com/mainflux/mainflux/auth"
+	"github.com/mainflux/mainflux/pkg/errors"
+)
+
+type createGroupReq struct {
+	token       string
+	Name        string                 `json:"name,omitempty"`
+	ParentID    string                 `json:"parent_id,omitempty"`
+	Description string                 `json:"description,omitempty"`
+	Metadata    map[string]interface{} `json:"metadata,omitempty"`
+}
+
+func (req createGroupReq) validate() error {
+	if req.token == "" {
+		return auth.ErrUnauthorizedAccess
+	}
+	if len(req.Name) > maxNameSize || req.Name == "" {
+		return errors.Wrap(auth.ErrMalformedEntity, auth.ErrBadGroupName)
+	}
+
+	return nil
+}
+
+type updateGroupReq struct {
+	token       string
+	id          string
+	Name        string                 `json:"name,omitempty"`
+	Description string                 `json:"description,omitempty"`
+	Metadata    map[string]interface{} `json:"metadata,omitempty"`
+}
+
+func (req updateGroupReq) validate() error {
+	if req.token == "" {
+		return auth.ErrUnauthorizedAccess
+	}
+
+	if req.id == "" {
+		return auth.ErrMalformedEntity
+	}
+
+	return nil
+}
+
+type listGroupsReq struct {
+	token string
+	id    string
+	level uint64
+	// - `true`  - result is JSON tree representing groups hierarchy,
+	// - `false` - result is JSON array of groups.
+	tree     bool
+	metadata auth.GroupMetadata
+}
+
+func (req listGroupsReq) validate() error {
+	if req.token == "" {
+		return auth.ErrUnauthorizedAccess
+	}
+
+	if req.level > auth.MaxLevel && req.level < auth.MinLevel {
```

Comment:
These should be two separate cases(and errors). If you want to use a single error, it needs to be `OR` rather than `AND`.